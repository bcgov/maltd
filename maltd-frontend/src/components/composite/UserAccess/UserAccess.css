th {
  font-weight: normal;
}

.border-bottom {
  border-bottom: "1px black solid";
}
