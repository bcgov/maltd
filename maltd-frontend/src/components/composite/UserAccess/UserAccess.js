import React from "react";
import PropTypes from "prop-types";
import { Container, Row, Col } from "reactstrap";
import ListElement from "../../base/ListElement/ListElement";
import PlusIcon from "../../base/PlusIcon/PlusIcon";
import Dropdown from "../../base/Dropdown/Dropdown";
import "./UserAccess.css";

export default function UserAccess({
  userAccess: { projects, userName, userEmail },
  onXClick,
  onPlusClick,
  onDropdownClick,
  dropdown,
  duplicateErrorMessage
}) {
  const userListElement = {
    title: userName,
    description: userEmail
  };
  const projectExists =
    projects && projects.length > 0 ? "projects" : "noProjects";
  let key = null;

  return (
    <Container className="move-down">
      <Row xs="1" sm="1" md="2" id="user-access-row">
        <Col className="cols">
          <Row className="inner-row">
            <Col className="big-font">USER</Col>
          </Row>
          <ListElement listElement={userListElement} />
          {duplicateErrorMessage && (
            <small className="error-message">{duplicateErrorMessage}</small>
          )}
        </Col>
        <Col className="top-projects" id={projectExists}>
          <div className="cols">
            <Row>
              <Col className="big-font">CURRENT PROJECTS</Col>
            </Row>
            {projectExists === "projects" &&
              projects.map(value => {
                const listElement = {
                  title: value.name,
                  resources: value.resources,
                  id: value.id
                };

                key = listElement.id;

                return (
                  <div key={key}>
                    <ListElement
                      listElement={listElement}
                      onXClick={onXClick}
                      id={key}
                    />
                  </div>
                );
              })}
            {(!projects || projects.length === 0) && (
              <ListElement listElement={{}} />
            )}
          </div>
        </Col>
      </Row>
      <Row xs="1" sm="1" md="2" className="outer">
        <Col />
        <Col>
          {projects && projects.length > 0 && (
            <div className="drop-plus" key={key}>
              <Dropdown dropdown={dropdown} onDropdownClick={onDropdownClick} />
              {dropdown.selectedDropdownItem && (
                <PlusIcon onClick={onPlusClick} />
              )}
            </div>
          )}
          {(!projects || projects.length === 0) && (
            <React.Fragment>
              <div className="drop-plus">
                <Dropdown
                  dropdown={dropdown}
                  onDropdownClick={onDropdownClick}
                />
                {dropdown.selectedDropdownItem && (
                  <PlusIcon onClick={onPlusClick} />
                )}
              </div>
            </React.Fragment>
          )}
        </Col>
      </Row>
    </Container>
  );
}

UserAccess.propTypes = {
  userAccess: PropTypes.shape({
    projects: PropTypes.array.isRequired,
    userName: PropTypes.string.isRequired,
    userEmail: PropTypes.string
  }).isRequired,
  onXClick: PropTypes.func.isRequired,
  onPlusClick: PropTypes.func.isRequired,
  onDropdownClick: PropTypes.func.isRequired,
  dropdown: PropTypes.shape({
    items: PropTypes.array.isRequired,
    selectedDropdownItem: PropTypes.object
  }).isRequired,
  duplicateErrorMessage: PropTypes.string.isRequired
};
