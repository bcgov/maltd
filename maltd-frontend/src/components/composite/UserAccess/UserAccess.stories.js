/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import UserAccess from "./UserAccess";
import "./UserAccess.css";

const userAccess = {
  projects: [{ name: "Project1" }, { name: "Project2" }],
  userName: "User Name",
  userEmail: "user@gov.bc.ca"
};

storiesOf("UserAccess", module)
  .add("default", () => <UserAccess userAccess={userAccess} />)
  .add("no projects", () => (
    <UserAccess userAccess={{ ...userAccess, projects: [] }} />
  ));
