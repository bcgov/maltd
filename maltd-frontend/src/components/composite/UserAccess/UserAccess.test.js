import React from "react";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import UserAccess from "./UserAccess";

Enzyme.configure({ adapter: new Adapter() });

describe("User Access", () => {
  const userName = "username";
  const userEmail = "useremail@gov.bc.ca";

  test("Component renders as expected", () => {
    const component = renderer.create(
      <UserAccess
        userAccess={{
          projects: [],
          userName,
          userEmail
        }}
        onXClick={() => jest.fn()}
        onPlusClick={() => jest.fn()}
        onDropdownClick={() => jest.fn()}
        dropdown={{
          items: [{ id: "123", name: "name", type: "type" }]
        }}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Component renders with no projects text when there are no projects for a user", () => {
    const component = shallow(
      <UserAccess
        userAccess={{
          projects: [],
          userName,
          userEmail
        }}
        onXClick={() => jest.fn()}
        onPlusClick={() => jest.fn()}
        onDropdownClick={() => jest.fn()}
        dropdown={{
          items: [{ id: "123", name: "name", type: "type" }]
        }}
      />
    );

    const rows = component.find("#user-access-row");
    expect(rows.length).toEqual(1);
    expect(
      rows
        .first()
        .find("Col")
        .find("#noProjects").length
    ).toEqual(1);
  });

  test("Component renders with existing projects when there are projects for a user", () => {
    const component = shallow(
      <UserAccess
        userAccess={{
          projects: [
            { name: "Project1", id: "123" },
            { name: "Project2", id: "1234" }
          ],
          userName,
          userEmail
        }}
        onXClick={() => jest.fn()}
        onPlusClick={() => jest.fn()}
        onDropdownClick={() => jest.fn()}
        dropdown={{
          items: [{ id: "123", name: "name", type: "type" }]
        }}
      />
    );

    const rows = component.find("#user-access-row");
    expect(rows.length).toEqual(1);
    expect(
      rows
        .first()
        .find("Col")
        .find("#projects").length
    ).toEqual(1);
  });
});
