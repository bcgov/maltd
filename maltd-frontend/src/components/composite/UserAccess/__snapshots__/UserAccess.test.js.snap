// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Access Component renders as expected 1`] = `
<div
  className="move-down container"
>
  <div
    className="row row-cols-1 row-cols-sm-1 row-cols-md-2"
    id="user-access-row"
  >
    <div
      className="cols col"
    >
      <div
        className="inner-row row"
      >
        <div
          className="big-font col"
        >
          USER
        </div>
      </div>
      <div
        className="project-div"
      >
        <div
          className="project-list-item"
          id="project-info"
        >
          <strong
            className="large-size"
          >
            username
          </strong>
          <p
            className="project-list-item"
            id="email-info"
          >
            useremail@gov.bc.ca
          </p>
          
          
        </div>
        <div />
      </div>
    </div>
    <div
      className="top-projects col"
      id="noProjects"
    >
      <div
        className="cols"
      >
        <div
          className="row"
        >
          <div
            className="big-font col"
          >
            CURRENT PROJECTS
          </div>
        </div>
        <div>
          <p>
            No projects
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="outer row row-cols-1 row-cols-sm-1 row-cols-md-2"
  >
    <div
      className="col"
    />
    <div
      className="col"
    >
      <div
        className="drop-plus"
      >
        <div
          className="dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            aria-label={null}
            className="btn btn-secondary"
            id="dropdown"
            onClick={[Function]}
            type="button"
          >
            <div
              id="title"
            >
              Select a project
            </div>
            <div
              className="img"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                />
              </svg>
            </div>
          </button>
          <div
            aria-hidden={true}
            className="dropdown-menu"
            data-cy="drop-down-menu"
            role="menu"
            tabIndex="-1"
          >
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              name
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      
    </div>
  </div>
</div>
`;
